pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -143,14 +145,20 @@ static boolean isCursorFromReplicatedSubscription(ManagedCursor cursor) {
     }
 
     public PersistentSubscription(PersistentTopic topic, String subscriptionName, ManagedCursor cursor,
-            boolean replicated) {
+                                                         boolean replicated) {
+        this(topic, subscriptionName, cursor, replicated, Collections.emptyMap());
+    }
+
+    public PersistentSubscription(PersistentTopic topic, String subscriptionName, ManagedCursor cursor,
+            boolean replicated, Map<String, Long> subscriptionProperties) {
         this.topic = topic;
         this.cursor = cursor;
         this.topicName = topic.getName();
         this.subName = subscriptionName;
         this.fullName = MoreObjects.toStringHelper(this).add("topic", topicName).add("name", subName).toString();
         this.expiryMonitor = new PersistentMessageExpiryMonitor(topicName, subscriptionName, cursor, this);
         this.setReplicated(replicated);
+        this.subscriptionProperties = subscriptionProperties == null ? new HashMap<>() : subscriptionProperties;
```

Comment:
```suggestion
        this.subscriptionProperties = subscriptionProperties == null ? Collections.emptyMap() : Collections.unmodifiableMap(subscriptionProperties);
```

Reply:
Collections.emptyMap() is a static Map, and data may be placed behind this Map, so I think it is better to new